packages/mui-base/src/Dialog/Root/useDialogRoot.ts

```diff
@@ -66,6 +68,12 @@ export function useDialogRoot(parameters: useDialogRoot.Parameters): useDialogRo
     };
   }, [open, onNestedDialogClose, onNestedDialogOpen, ownNestedOpenDialogs]);
 
+  React.useEffect(() => {
+    if (!open) {
+      setOpenMethod(null);
+    }
+  }, [open]);
```

Comment:
If this needs to be sync'd via an effect, this is better:

```suggestion
   if (!open && openMethod !== null) {
     setOpenMethod(null)
   }
```